src/components/Warning/warning.tsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+type Props = {
+  message: string;
+};
+
+const Warning: React.FC<Props> = ({ message }) => {
+  const [isShown, setIsShown] = useState(false);
+
+  useEffect(() => {
+    if (message) {
+      setIsShown(true);
+      setTimeout(() => {
+        setIsShown(false);
+      }, 3000);
+    } else {
+      setIsShown(false);
+    }
+  }, [message]);
```

Comment:
```suggestion
  useEffect(() => {
    let timer;
    
    if (message) {
      setIsShown(true);
      timer = setTimeout(() => {
        setIsShown(false);
      }, 3000);
    } else {
      setIsShown(false);
    }
    
    return () => {
      if (timer) {
        clearTimeout(timer);
      }
    }
  }, [message]);
```

You should clear any async timeouts in useEffect callback function.